app/upgrades/v15rc2/upgrades.go

```diff
@@ -6,19 +6,49 @@ package v15rc2
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/evmos/evmos/v15/utils"
 )
 
 // CreateUpgradeHandler creates an SDK upgrade handler for v15.0.0-rc2
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
+	ak authzkeeper.Keeper,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 		logger := ctx.Logger().With("upgrade", UpgradeName)
 
+		if utils.IsTestnet(ctx.ChainID()) {
+			logger.Info("removing outdated distribution authorizations")
+			RemoveDistributionAuthorizations(ctx, ak)
+		}
+
 		// Leave modules are as-is to avoid running InitGenesis.
 		logger.Debug("running module migrations ...")
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
+
+// RemoveDistributionAuthorizations removes any outdated distribution authorizations
+// found in the authz keeper.
+func RemoveDistributionAuthorizations(ctx sdk.Context, ak authzkeeper.Keeper) {
+	ak.IterateGrants(ctx, func(granterAddr, granteeAddr sdk.AccAddress, grant authz.Grant) bool {
+		authorization, err := grant.GetAuthorization()
+		if err != nil {
+			return true
```

Comment:
this shouldn't stop the iteration
```suggestion
			return false
```